test_compression_column_tforms failure with cfitsio 3.380

With the latest relase of cfitsio (which just entered Debian unstable), there are two test failures. The first one:
```
_________ TestCompressedImage.test_compression_column_tforms ______________

self = <astropy.io.fits.tests.test_image.TestCompressedImage object at 0x7fcf39d96210>

    def test_compression_column_tforms(self):
        """Regression test for https://aeon.stsci.edu/ssb/trac/pyfits/ticket/199"""
    
        # Some interestingly tiled data so that some of it is quantized and
        # some of it ends up just getting gzip-compressed
        data2 = ((np.arange(1, 8, dtype=np.float32) * 10)[:, np.newaxis] +
                 np.arange(1, 7))
        np.random.seed(1337)
        data1 = np.random.uniform(size=(6 * 4, 7 * 4))
        data1[:data2.shape[0], :data2.shape[1]] = data2
        chdu = fits.CompImageHDU(data1, compressionType='RICE_1',
                                 tileSize=(6, 7))
        chdu.writeto(self.temp('test.fits'))
    
        with fits.open(self.temp('test.fits'),
                       disable_image_compression=True) as h:
>           assert h[1].header['TFORM1'] == '1PB(30)'
E           assert '1PB(45)' == '1PB(30)'
E             - 1PB(45)
E             + 1PB(30)

astropy/io/fits/tests/test_image.py:1268: AssertionError
```
This may be a bug in cfitsio as well; I would need your hint here.
Full log [here](https://buildd.debian.org/status/fetch.php?pkg=python-astropy&arch=amd64&ver=1.1.1-3%2Bb1&stamp=1456707318).
This happens on all architectures.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -104,6 +104,9 @@ Bug Fixes
 
 - ``astropy.io.fits``
 
+  - Made TFORMx keyword check more flexible in test of compressed images to
+    enable copatibility of the test with cfitsio 3.380. [#4646]
+
 - ``astropy.io.misc``
 
 - ``astropy.io.registry``

--- a/astropy/io/fits/tests/test_image.py
+++ b/astropy/io/fits/tests/test_image.py
@@ -4,6 +4,7 @@ from __future__ import division, with_statement
 
 import math
 import os
+import re
 import time
 import warnings
 
@@ -1265,8 +1266,8 @@ class TestCompressedImage(FitsTestCase):
 
         with fits.open(self.temp('test.fits'),
                        disable_image_compression=True) as h:
-            assert h[1].header['TFORM1'] == '1PB(30)'
-            assert h[1].header['TFORM2'] == '1PB(359)'
+            assert re.match(r'^1PB\(\d+\)$', h[1].header['TFORM1'])
+            assert re.match(r'^1PB\(\d+\)$', h[1].header['TFORM2'])
 
     def test_compression_update_header(self):
         """Regression test for

